pkg/server/commands/write_assertions_test.go

```diff
@@ -161,4 +165,350 @@ func TestWriteAssertions(t *testing.T) {
 		require.Nil(t, resp)
 		require.ErrorIs(t, err, serverErrors.ExceededEntityLimit("bytes", DefaultMaxAssertionSizeInBytes))
 	})
+
+	storeID := ulid.Make().String()
+	modelID := ulid.Make().String()
+
+	model := parser.MustTransformDSLToProto(`
+	model
+		schema 1.1
+	type user
+
+	type repo
+		relations
+			define reader: [user, user with condX]
+			define can_read: reader
+
+	condition condX(x :int) {
+		x > 0
+	}`)
+
+	modelInvalidVersion := &openfgav1.AuthorizationModel{
+		SchemaVersion: "1.0",
+	}
+
+	var tests = []struct {
+		_name         string
+		input         *openfgav1.WriteAssertionsRequest
+		setMock       func(*mockstorage.MockOpenFGADatastore)
+		expectedError string
+	}{
+		{
+			_name: "succeeds",
+			input: &openfgav1.WriteAssertionsRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Assertions: []*openfgav1.Assertion{
+					{
+						TupleKey:    tuple.NewAssertionTupleKey("repo:test", "reader", "user:elbuo"),
+						Expectation: false,
+					},
+					{
+						TupleKey:    tuple.NewAssertionTupleKey("repo:test", "reader", "user:maria"),
+						Expectation: true,
+					},
+					{
+						TupleKey:    tuple.NewAssertionTupleKey("repo:test", "reader", "user:jon"),
+						Expectation: false,
+					},
+					{
+						TupleKey:    tuple.NewAssertionTupleKey("repo:test", "reader", "user:jose"),
+						Expectation: true,
+					},
+				},
+			},
+			setMock: func(mockDatastore *mockstorage.MockOpenFGADatastore) {
+				mockDatastore.EXPECT().ReadAuthorizationModel(gomock.Any(), storeID, modelID).Times(1).Return(model, nil)
+				mockDatastore.EXPECT().WriteAssertions(gomock.Any(), storeID, modelID, gomock.Any()).Times(1).Return(nil)
+			},
+		},
+		{
+			_name: "succeeds_when_empty_assertions",
```

Comment:
Any reason for the `_` preceding `_name`?

Reply:
No reason, updated :)